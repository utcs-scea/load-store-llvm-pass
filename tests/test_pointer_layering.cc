#include <stdio.h>

int global_a = 10;
int global_b = 20;
int *global_ptr_a = &global_a;
int *global_ptr_b = &global_b;

void swap_pointers() {
  int *temp = global_ptr_a;
  global_ptr_a = global_ptr_b;
  global_ptr_b = temp;
}

void modify_through_pointers() {
  *global_ptr_a += 5;
  *global_ptr_b *= 2;
}

int main() {
  printf("Initial values: global_a = %d, global_b = %d\n", global_a, global_b);
  printf("Initial pointer values: *global_ptr_a = %d, *global_ptr_b = %d\n", *global_ptr_a, *global_ptr_b);

  modify_through_pointers();
  printf("After modification: global_a = %d, global_b = %d\n", global_a, global_b);

  swap_pointers();
  printf("After swapping pointers: *global_ptr_a = %d, *global_ptr_b = %d\n", *global_ptr_a, *global_ptr_b);

  modify_through_pointers();
  printf("Final values: global_a = %d, global_b = %d\n", global_a, global_b);

  return 0;
}
