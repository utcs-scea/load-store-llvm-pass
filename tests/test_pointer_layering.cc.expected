   >> globalify() invoked
   >> globalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
Initial values: global_a = 10, global_b = 20
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
Initial pointer values: *global_ptr_a = 10, *global_ptr_b = 20
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
After modification: global_a = 15, global_b = 40
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
After swapping pointers: *global_ptr_a = 40, *global_ptr_b = 15
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_ptr() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
Final values: global_a = 30, global_b = 45
